src/views/players/PlayerView/PlayerView.vue

```diff
@@ -224,16 +289,68 @@
 
     isAchieved(a: ProfileAchievement): boolean {
       let b = true;
-      Object.values(this.myAchievements).forEach( (value) => {
-        if (value.title === a.title) {
+      Object.values(this.myAchievements).forEach((value) => {
+        if (!Object.prototype.hasOwnProperty.call(a, 'title')) {
+          Object.values(a).forEach((value2) => {
+            if (value.title === value2.title) {
+              b = false;
+            }
+          });
+        } else if (value.title === a.title) {
           b = false;
         }
       });
       return b;
     }
+
+    completed(a: ProfileAchievement[]): number {
+      let b = 0;
+      Object.values(this.myAchievements).forEach((value) => {
+        if (!Object.prototype.hasOwnProperty.call(a, 'title')) {
+          Object.values(a).forEach((value2) => {
+            if (value.title === value2.title) {
+              b += 1;
+            }
+          });
+        }
+      });
+      return b;
+    }
+
+    computeAchievement(a: ProfileAchievement): ProfileAchievement {
+      if (!Object.prototype.hasOwnProperty.call(a, 'title')) {
+        let highest = this.completed(Object.values(a));
+        if (highest === 0){
+          highest = 1;
+        }
```

Comment:
```suggestion
        const highest = this.completed(Object.values(a)) || 1;
```
Nit, but I think this condensed logic is a bit cleaner (also lets us avoid mutating `highest`)